api/src/services/settings.js

```diff
@@ -55,6 +57,9 @@ module.exports = {
         if (!doc.settings) {
           doc.settings = {};
         }
+
+        _.defaults(body.permissions, defaults.permissions);
```

Comment:
I think this should be done after the replace/extend, ie: just before the put line 68. Then you can do the defaulting straight on to `doc.settings` which most of the time won't do anything.